docs/development/howto.rst

```diff
@@ -907,6 +907,29 @@ putting this in ``docs/conf.py`` can also help sometimes::
     warnings.showwarning = warn_with_traceback
 
 
+Providing summary info for class instances
```

Comment:
Maybe a shorter section title that fits in the sidebar?
E.g. "Object summary info string"?